src/main/java/org/wildfly/security/auth/server/RealmIdentity.java

```diff
@@ -148,6 +149,17 @@ default AuthorizationIdentity getAuthorizationIdentity() throws RealmUnavailable
     }
 
     /**
+     * Get the attributes for the realm identity.
+     *
+     * @return the attributes
+     * @throws UnsupportedOperationException if the implementing class does not support getting attributes
+     * @throws RealmUnavailableException if accessing the attributes fails for some reason
+     */
+    default Attributes getAttributes() throws RealmUnavailableException {
```

Comment:
Wouldn't it be better to return null?

Despite the number forgotten I think it is easier to remember to check for null than it is to remember to catch an unchecked exception after reading the javadoc.